### PR TITLE
VATRP-3991: Tablet contact crash fix

### PR DESCRIPTION
**Pull request by samson-accontech**

### CODE REVIEW

File: src/org/linphone/setup/CDNProviders.java [hunk 1]
```diff
@@ -140,62 +141,72 @@ public boolean updateProviders(String jsonString, boolean save) {
...
+				tmp = new Provider();
+				try {
 
-				tmp.name = ((JSONObject) jsonArray.get(i)).getString("name");
-				tmp.icon = ((JSONObject) jsonArray.get(i)).getString("icon");
-				tmp.icon2x = ((JSONObject) jsonArray.get(i)).getString("icon2x");
-				//tmp.port=srvLookup
+					//change domains to valid domains, if not listed on the server that way
+					if (((JSONObject) jsonArray.get(i)).getString("domain").equals("zvrs.vatrp.net")) {
+						tmp.domain = "208.94.16.87";
```
Comment (zackm0571) [inline src/org/linphone/setup/CDNProviders.java:156]: Get rid of this. This is hardcoded and pulled from the CDN. Should not be modified in app.

File: src/org/linphone/setup/CDNProviders.java [hunk 2]
```diff
@@ -140,62 +141,72 @@ public boolean updateProviders(String jsonString, boolean save) {
...
+					} else if (((JSONObject) jsonArray.get(i)).getString("domain").equals("psip-lb.staging.purple.us")) {
+						tmp.domain = "psip-lb.test.purple.us";
+					} else {
```
Comment (zackm0571) [inline src/org/linphone/setup/CDNProviders.java:159]: Get rid of this

File: src/org/linphone/setup/CDNProviders.java (hunk 1 above)
Comment (Semrubo) [inline src/org/linphone/setup/CDNProviders.java:156]: This was not changed, it was hardcoded before. If you open the file you can see same lines in previous code. The github diff incorrectly shows this part changed.

File: src/org/linphone/setup/CDNProviders.java (hunk 2 above)
Comment (Semrubo) [inline src/org/linphone/setup/CDNProviders.java:159]: This was the same in previous code, incorrect comparison result

Comment (shareefalis) [inline src/org/linphone/setup/CDNProviders.java:159]: Looks fine to me

File: src/org/linphone/setup/CDNProviders.java (hunk 1 above)
Comment (shareefalis) [inline src/org/linphone/setup/CDNProviders.java:156]: looks fine to me

**Comment by shareefalis** [comment]
conflicts..

**Comment by patrickmwatson** [comment]
Closed moved to PUC

**Pull request closed without merging**